db/migrate/20160825172604_add_shared_to_agents.rb

```diff
@@ -0,0 +1,9 @@
+class AddSharedToAgents < ActiveRecord::Migration
+  class Agents < ActiveRecord::Base
```

Comment:
Is this necessary?  If it is, I think you actually mean`::Agents`, or it will define AddSharedToAgents::Agents if the class is not defined yet.